app/lib/classification-queue.js

```diff
@@ -0,0 +1,101 @@
+import apiClient from 'panoptes-client/lib/api-client';
+import { Split } from 'seven-ten';
+
+const FAILED_CLASSIFICATION_QUEUE_NAME = 'failed-classifications';
+const MAX_RECENTS = 10;
+
+class ClassificationQueue {
+    constructor(storage, api) {
+        this.storage = storage || window.localStorage;
+        this.apiClient = api || apiClient;
+        this.recents = [];
+    }
+
+    add(classification) {
+        var queue = this._loadQueue();
+        queue.push(classification);
+
+        try {
+            this._saveQueue(queue);
+            console.info('Queued classifications:', queue.length);
+        }
+        catch(error) {
+            console.error('Failed to queue classification:', error);
+        }
+
+        this.flushToBackend();
+    }
+
+    length() {
+        return this._loadQueue().length
+    }
+
+    addRecent(classification) {
+        if (this.recents.length > MAX_RECENTS) {
+            let droppedClassification = this.recents.pop();
+            droppedClassification.destroy();
+        }
+
+        this.recents.unshift(classification);
+    }
+
+    flushToBackend() {
+        var queue = this._loadQueue();
+
+        if (queue.length > 0) {
+            console.log('Saving queued classifications:', queue.length);
+            for (let classificationData of queue) {
+                this.apiClient.type('classifications').create(classificationData)
+                    .save()
+                    .then((actualClassification) => {
+                        console.log('Saved classification', actualClassification.id);
+                        Split.classificationCreated(actualClassification); // Metric log needs classification id
```

Comment:
We should find out if Split is still needed here.

Reply:
Actually, would it be better to pass a post-classification callback to the constructor, invoke it here and leave `Split` in the calling page?

Reply:
The metric logs Seven Ten makes needs the classification id if the split is tracking the classification created to find out if the split has an effect on classification. The researcher can then compare the split assigned to the user to the classification data export using the id.

Reply:
A callback would be fine for that and I think preferable long term.